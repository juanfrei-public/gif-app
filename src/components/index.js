export * from './AddCategory';
export * from './Item';
export * from './ListarItems';
export * from './Title';
export * from './Loading';