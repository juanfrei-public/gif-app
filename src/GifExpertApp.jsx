import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { ListarItems } from "./components/ListarItems";
import { Title } from "./components/Title";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Bleach"]);

  const onAddCategory = (value) => {
    if (categories.includes(value)) return;

    // Formas de agregar categorias
    setCategories([...categories, value]);
    // setCategories( cat => [...cat, value]);
  };

  return (
    <>
      <Title titulo={"Expert gif app"} />

      <AddCategory
        // onAddCategory={onAddCategory} - se envia metodo al hijo para agregar categoria (no es lo mejor en este caso)
        onNewCategory={(value) => onAddCategory(value)}
      />

      {/* Pre componente */}
      {/* <ol>
        {categories.map((category) => (
          <div key={category}>
            <h3>{category}</h3>
            <li>{category}</li>
          </div>
        ))}
      </ol> */}

      {/* Listado en componente */}
      {categories.map((category) => (
        <ListarItems key={category} items={category} />
      ))}
    </>
  );
};
