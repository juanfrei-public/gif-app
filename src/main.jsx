import React from 'react';
import ReactDOM from 'react-dom/client';
import { GifExpertApp } from './GifExpertApp';

import './styles.css';

// LOBcuGYUNa5QAVhlG1zCeOKCaFv5T7Qk API KEY


ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExpertApp/>
  </React.StrictMode>
)
