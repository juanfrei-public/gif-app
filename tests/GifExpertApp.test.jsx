import {
  fireEvent,
  render,
  screen,
  waitFor,
  renderHook,
} from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("Pruebas sobre GifExperApp", () => {
  test("Debe matchear con el snapshot", () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });

  test("Debe tener un titulo en etiqueta h1", () => {
    const titleInHeading = "Expert gif app";
    const tagType = "h1";
    render(<GifExpertApp />);
    const { tagName } = screen.getByRole("heading");
    expect(tagName.toLowerCase()).toBe(tagType);
    expect(screen.getByText(titleInHeading)).toBeTruthy();
  });

  xtest("Debe agregarse una categoria", async () => {
    const itemMock = "asdf";
    render(<GifExpertApp />);
  });
});
