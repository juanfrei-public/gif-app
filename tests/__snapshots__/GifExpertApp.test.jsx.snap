// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifExperApp Debe matchear con el snapshot 1`] = `
<div>
  <h1>
    Expert gif app
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
